File: criptoPython/AES-GCM-Cifrado.py
from Crypto.Cipher import AES

#Let´s declare a message to encrypt
message = bytes('Toda its a great day to start encryptying some maessage under an AES cipher and GCM mode', 'utf8')

#Declare our key in hex
key = bytes.fromhex('c936108299307d3f6f7585b96013346d')
#Declare our nonce in hex
nonce = bytes.fromhex('47e6831df094b7a6')

#Declare our associate data
associate_data = bytes('Today is tursday','utf8')

#Lets go and cipher
cipher = AES.new(key, AES.MODE_GCM, nonce=nonce)
#Update the cipher usin our associate datta
cipher.update(associate_data)
#Cipher and authenticate
cipher_text, mac = cipher.encrypt_and_digest(message)

#Pu it on the screen
print("Cipher_text is: ", cipher_text.hex())
print("Tag is: ", mac.hex())

File: CyberSecurity-Py/criptoGo/AES-GCM-Cifrado.go
package main

import (
	"crypto/aes"
	"crypto/cipher"
	"crypto/rand"
	"encoding/hex"
	"fmt"
	"io"
)

func main() {
	//We start with the message tpo encrypt
	message := []byte("Today its a great day to start encrypting some message under Golang Languaje")

	//We declare our hexadecimal key
	key, _ := hex.DecodeString("c936108299307d3f6f7585b96013346d")

	//We declare our nonce and put in it a try catch sentence to catch any errors
	nonce := make([]byte, 12)
	if _, err := io.ReadFull(rand.Reader, nonce); err != nil {
		panic(err.Error)
	}

	// We start our encrypt in AES algorithm with GCM mode
	block, err := aes.NewCipher(key)
	if err != nil {
		panic(err.Error())
	}

	aesgcm, err := cipher.NewGCM(block)
	if err != nil {
		panic(err.Error())
	}

	// Encrypt and authenticate
	ciphertext := aesgcm.Seal(nil, nonce, message, nil)

	// Show the encrypted and MAC id!!!
	fmt.Printf("Cipher_text is: %x\n", ciphertext[:len(ciphertext)-aesgcm.Overhead()])
	fmt.Printf("Tag is: %x\n", ciphertext[len(ciphertext)-aesgcm.Overhead():])
}
